packages/teams-js/src/public/runtime.ts

```diff
@@ -439,6 +439,42 @@ export const mapTeamsVersionToSupportedCapabilities: Record<string, Array<ICapab
 };
 
 const generateBackCompatRuntimeConfigLogger = runtimeLogger.extend('generateBackCompatRuntimeConfig');
+
+/**
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * Merges the capabilities of two runtime objects. Fully supports arbitrarily nested capabilities/subcapabilities.
+ *
+ * Note that the two parameters are named "baselineRuntime" and "runtimeToMergeIntoBaseline" to make it easier
+ * to understand what this function is doing -- but the order of the parameters does not matter, it's just merging
+ * the two runtime objects together.
+ *
+ * Also note that this function isn't actually doing anything specific to capabilities/runtime. It's just doing a
+ * generic merge of two objects.
+ *
+ * @param baselineRuntime the baseline runtime object
+ * @param runtimeToMergeIntoBaseline the runtime object to merge into the baseline
+ * @returns the merged runtime object
+ */
+function mergeRuntimeCapabilities(baselineRuntime: object, runtimeToMergeIntoBaseline: object): object {
+  const merged: object = { ...baselineRuntime };
+
+  for (const key in runtimeToMergeIntoBaseline) {
+    if (Object.prototype.hasOwnProperty.call(runtimeToMergeIntoBaseline, key)) {
```

Comment:
The `Object.prototype.hasOwnProperty.call` method is used to check if an object has a property as its own property (as opposed to inheriting it from its prototype chain). This method is preferred over calling the `hasOwnProperty` method on the object itself because it is possible for the object to have a property with the name `hasOwnProperty` that shadows the method from `Object.prototype`. This can cause unexpected behavior or even errors if the `hasOwnProperty` property on the object is not a function. By calling the method from `Object.prototype` directly, we can ensure that we are using the correct method to check for the object’s own properties.